aiohttp/multipart.py

```diff
@@ -636,7 +652,24 @@ async def next(
             await self._read_boundary()
         if self._at_eof:  # we just read the last boundary, nothing to do there
             return None
-        self._last_part = await self.fetch_next_part()
+
+        part = await self.fetch_next_part()
+        # https://datatracker.ietf.org/doc/html/rfc7578#section-4.6
+        if (
+            self._last_part is None
+            and self._mimetype.subtype == "form-data"
+            and isinstance(part, BodyPartReader)
+        ):
+            _, params = parse_content_disposition(part.headers.get(CONTENT_DISPOSITION))
+            if params.get("name") == "_charset_":
+                # Longest encoding in https://encoding.spec.whatwg.org/encodings.json
+                # is 19 characters, so 32 should be more than enough for any valid encoding.
+                charset = await part.read_chunk(32)
+                if len(charset) >= 32:
```

Comment:
Is `read_chunk(32)` able to return more than 32 bytes? I thought that argument is the max size...
Also, `>=` includes `==` so do you expect 31 bytes max?

Reply:
> Is `read_chunk(32)` able to return more than 32 bytes? I thought that argument is the max size...

No, I just always write such comparisons defensively. Can be changed to `==` if preferred.

> Also, `>=` includes `==` so do you expect 31 bytes max?

As per the comment, I expect something closer to 19 bytes max, just picked the next power of 2 to give it some leeway. e.g. Maybe it includes the line endings, which would make it 21 bytes. Maybe somebody includes some whitespace characters that pushes it to 22-24 bytes or similar.

I just needed to pick a value to avoid any DoS issue with reading an entire chunk into memory. This should be more than enough for anything valid.

Reply:
It probably makes sense to put `32` into a constant and reuse it.

It's just confusing to read that you first request maximum 32 bytes and then check if it's not more even though the called function should be limiting that already. Should the check be `>`?